docs/book/coupons.rst

```diff
@@ -0,0 +1,105 @@
+.. index::
+   single: Coupons
+
+Coupons
+=======
+
+The concept of Coupons in closely connected to the :doc:`Promotions Concept </book/promotions>`.
+
+Coupon Parameters
+-----------------
+
+A **Coupon** besides a ``code`` has a date when it expires, the ``usageLimit`` and it counts how many times it was already used.
+
+How to create a Coupon with a Promotion programmatically?
+---------------------------------------------------------
+
+.. warning::
+
+   The Promotion has to be ``couponBased = true`` in order to be able to hold a collection of Coupons that belong to it.
+
+Let's create a promotion that will have one coupon that activates the free shipping.
+
+.. code-block:: php
+
+   /** @var PromotionInterface $promotion */
+   $promotion = $this->container->get('sylius.factory.promotion')->createNew();
+
+   $promotion->setCode('free_shipping');
+   $promotion->setName('Free Shipping');
+   // Remember to set a channel for your promotion!
+   $promotion->addChannel($this->container->get('sylius.repository.channel')->findOneBy(['code' => 'US_Web_Store']));
+
+   $promotion->setCouponBased(true);
+
+   // Now create a Coupon and add it to
+   /** @var CouponInterface $coupon */
+   $coupon = $this->container->get('sylius.factory.promotion_coupon')->createNew();
+
+   $coupon->setCode('FREESHIPPING');
+
+   $promotion->addCoupon($coupon);
+
+   // Now create an Action that will take place after applying this promotion - 100% discount on shipping
+   /** @var ActionFactoryInterface $actionFactory */
+   $actionFactory = $this->container->get('sylius.factory.promotion_action');
+
+   // Provide the amount in float ( 1 = 100% )
+   $action = $actionFactory->createPercentageShippingDiscount(1);
+
+   $promotion->addAction($action);
+
+   $this->container->get('sylius.repository.promotion')->add($promotion);
+
+Finally to see the effects of your promotion with Coupon you need to **apply a coupon on the Order**.
+
+How to apply a Coupon to an Order?
+----------------------------------
+
+To apply you promotion with coupon that gives 100% discount on the shipping costs
+you need an order that has shipments. Set your promotion coupon on that order -
+this is what happens when a customer provides a coupon code during checkout.
+
+And after that call the OrderProcessor on the Order to have the promotion applied.
+
+.. code-block:: php
+
+   $order->setPromotionCoupon($coupon);
+
+   $this->container->get('sylius.order_processing.order_processor')->process($order);
+
+Coupon Generator
+----------------
+
+Making up new codes might become difficult if you would like to generate a lot of coupons at once. That is why Sylius
```

Comment:
`...if you would like to create/make a lot...`
Sentence `if you would like to generate... use the generator`l, might hit on readers' intelligence in a `no shit sherlock` way